pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/TNCuda.hpp

```diff
@@ -252,6 +252,31 @@ class TNCuda : public TNCudaBase<PrecisionT, Derived> {
                                                         host_data_size);
     }
 
+    /**
+     * @brief Check if the provided MPS has the correct dimension for C++
+     * backend.
+     *
+     * @param MPS_shape_source Dimension list of incoming MPS.
+     */
+
+    void MPSShapeCheck(
+        const std::vector<std::vector<std::size_t>> &MPS_shape_source) {
+        bool sameShape = sitesExtents_ == MPS_shape_source;
+        if (!sameShape) {
+            auto MPS_shape_source_str =
+                Pennylane::Util::vector2DToString<std::size_t>(
+                    MPS_shape_source);
+            auto MPS_shape_dest_str =
+                Pennylane::Util::vector2DToString<std::size_t>(sitesExtents_);
```

Comment:
Convert the MPS shape array to string to print together with `PL_ABORT_IF_NOT` and give a hit to the user in case of failure.